activerecord/lib/active_record/timestamp.rb

```diff
@@ -124,6 +126,8 @@ def _update_record
         end
       end
 
+      return yield if block_given? && @called_already
```

Comment:
Am I right in saying that the point here is to have the implementation run twice?

If so I think there's probably a more appropriate name for the ivar.

Reply:
Did away with the ivar altogether for what I think is a cleaner implementation?: https://github.com/rails/rails/compare/a25a55b1ac11a7de8ca8584e278d78ee26f63d35..ba69cd2d2130cf6d42bd201da9b67ebd756ec4a4.